dids/src/main/kotlin/web5/sdk/dids/DidMethod.kt

```diff
@@ -24,7 +24,7 @@ import web5.sdk.crypto.KeyManager
  * Implementers should adhere to the respective DID method specifications ensuring both compliance
  * and interoperability across different DID networks.
  */
-public abstract class Did(public val uri: String, public val keyManager: KeyManager) {
+public abstract class StatefulDid(public val uri: String, public val keyManager: KeyManager) {
```

Comment:
What's the thinking behind "stateful"? Is is describing that key managers have state?

Reply:
i had the same q!

Reply:
also does this imply there's such a thing as a stateless did as well

Reply:
Yeah, basically based on the did-core spec (https://www.w3.org/TR/did-core/#terminology), `DID` itself just refers to the identifier, but our concept of `Did` contained the key manager + other things. And it's not immediately clear if we just say `Did` that it wraps the key manager (which you can also use independently of did proper operations)

Reply:
So it's a DID but also we have access to the key material. A DID that we "own".

Stateful is a bit confusing for me. Makes me think it's mutable, or not thread safe. Can't think of an obviously better name though. Best I've got is `DidResource`

Reply:
Depending on the DID method associated with a did, the DID Document associated with it may be mutable. DID ION, DID Web, and DID DHT are all example where that's the case.

Because of this, they aren't thread safe either.

Reply:
StatefulDid is a buzzkill on the naming to me..
could we keep the name Did but have like a property that gives the legit did without stateful stuff :hmm:

Reply:
`StatefulDid` would continue to be confusing to me if we hadn't had a long discussion earlier this week. I still think `Did` is the right way to go. not sure that people will be thinking about the definition of `Did` strictly as per the spec. 

`StatefulDid` makes me think the concept of a `StatelessDid` exists as well which would effectively just be the DID as a string.

Reply:
i think very few people are thinking about dids strictly as per spec and the likelihood of confusion caused by using the word `Stateful` is higher than the likelihood of confusion caused by not including it

Reply:
In accordance with the [spec terminology](https://www.w3.org/TR/did-core/#terminology) "DID" is technically just the "_globally unique persistent identifier_", which basically means the string-value `did:key:sdlgjhsgolwnglwjnrg`, but... all of the marketing material (for lack of a better term) is centered around "DID" which would require it being an abstraction which encapsulates everything else (keys, docs, resolution, etc) for the purposes of aligning public awareness ("marketing") with implementation.

Reply:
no stateful pls 😄 

if anything you might want to tag some DIDs as "self resolving" - which would just apply to did:key and did:jwk - but I'm not certain the value of that